numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -195,6 +197,41 @@ static void
 
 /**end repeat**/
 
+char *cross1d_signature = "(3),(3)->(3)";
+
+/**begin repeat
+
+   #TYPE=LONG,DOUBLE#
+   #typ=npy_long, npy_double#
+*/
+
+/*
+ *  This implements the cross product:
+ *        out[n, 0] = in1[n, 1]*in2[n, 2] - in1[n, 2]*in2[n, 1]
+ *        out[n, 1] = in1[n, 2]*in2[n, 0] - in1[n, 0]*in2[n, 2]
+ *        out[n, 2] = in1[n, 0]*in2[n, 1] - in1[n, 1]*in2[n, 0]
+ */
+static void
+@TYPE@_cross1d(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
```

Comment:
I may as well just ask--do we have some docs that describe the mechanics of the `@` symbol usage in NumPy C source? It *appears* to be some kind of mechanism for generalization / substitution, but unless I missed a chapter in the C text book, not standard?

Reply:
Yup, I know that now, but it is not well exposed documentation -- I've been working on #11689 & to be fair the distutils docs [template section](https://github.com/numpy/numpy/blob/master/doc/DISTUTILS.rst.txt#template-files) is nothing more than a placeholder.